bindings/java/native/src/classes/client/full_node_api.rs

```diff
@@ -0,0 +1,335 @@
+};
+
+use crate::{
+    address::*,
+    balance::GetBalanceBuilderApi,
+    bee_types::*,
+    client_builder::ClientBuilder,
+    message::{ClientMessageBuilder, GetMessageBuilder, MessageWrap, Message},
+    mqtt::MqttManager,
+};
+
+impl From<ClientRust> for Client {
+    fn from(client: ClientRust) -> Self {
+        Self(client)
+    }
+}
+
+pub struct Client(ClientRust);
+
+/// Full node API
+impl Client {
+    pub fn builder() -> ClientBuilder {
+        ClientBuilder::new()
+    }
+
+    pub fn borrow<'a>(&'a self) -> &'a ClientRust {
+        &self.0
+    }
+
+    pub fn borrow_mut<'a>(&'a mut self) -> &'a mut ClientRust {
+        &mut self.0
+    }
+
+    pub fn get_health(&self) -> Result<bool> {
+        Ok(crate::block_on(async { self.0.get_health().await })?)
+    }
+
+    pub fn get_node_health(&self, node: &str) -> Result<bool> {
+        Ok(crate::block_on(async {
+            iota_client::Client::get_node_health(node).await
+        })?)
+    }
+
+    pub fn get_info(&self) -> Result<NodeInfoWrapper> {
+        Ok(crate::block_on(async { self.0.get_info().await })?.into())
+    }
+
+    pub fn get_tips(&self) -> Result<Vec<String>> {
+        let tips = crate::block_on(async { self.0.get_tips().await })?;
+        Ok(tips.into_iter().map(|p| p.to_string()).collect())
+    }
+
+    pub fn get_peers(&self) -> Result<Vec<PeerDto>> {
+        Ok(crate::block_on(async { self.0.get_peers().await })?
+            .into_iter()
+            .map(PeerDto::from)
+            .collect())
+    }
+
+    pub fn post_message(&self, msg: Message) -> Result<MessageId> {
+        let ret = crate::block_on(async {
+            self.0.post_message(&msg.to_inner_clone()).await
+        });
+
+        match ret {
+            Ok(s) => Ok(s),
+            Err(e) => Err(anyhow!(e.to_string())),
+        }
+    }
+
+    pub fn get_output(&self, output_id: String) -> Result<OutputResponse> {
+        Ok(crate::block_on(async { self.0.get_output(&RustUtxoInput::from_str(&output_id)?).await })?.into())
+    }
+
+    /// GET /api/v1/addresses/{address} endpoint
+    pub fn get_address(&self) -> GetAddressBuilder {
+        GetAddressBuilder::new(self)
+    }
+
+    pub fn get_address_balance(&self, address: &str) -> Result<BalanceAddressResponse> {
+        let mut result: BalanceAddressResponse =
+            crate::block_on(async { self.0.get_address().balance(&String::from(address)).await })?.into();
+        result.address = crate::block_on(async { self.0.hex_to_bech32(&result.address, None).await })?;
+        Ok(result)
+    }
+
+    pub fn get_addresses_balances(&self, addresses: Vec<String>) -> Result<Vec<BalanceAddressResponse>> {
+        let result: Vec<BalanceAddressResponse> =
+            crate::block_on(async { self.0.get_address_balances(&addresses).await })?
+                .into_iter()
+                .map(|b| {
+                    let result: BalanceAddressResponse = b.into();
+                    // TODO
```

Comment:
What about this, do you want to do this in a later PR?

Reply:
Naee, can be removed. We do this in the python bindings, but we can also just use the hex_to_bech since i addded those in java